src/battle-dex.ts

```diff
@@ -128,6 +128,22 @@ function toUserid(text: any) {
 	return toId(text);
 }
 
+function parseNameParts(text: string) {
```

Comment:
I definitely don't think we need more globals.

I guess this could be stuck next to `parseLine`.

Reply:
I'd want this to be a static method on `User`, but `User` isn't accessible by `UserList`. Globals are the result of the current design in the client from what I can tell, and there's no suitable `Util` class to stick this under. As for `lineParse` - isn't that in the Preact code?

Reply:
There should be a line parser in `BattleLog`. Let me check.

Reply:
https://github.com/Zarel/Pokemon-Showdown-Client/blob/4d2f86be7dbf79f75513d2180c897cf8f0d8d09f/src/battle-text-parser.ts#L18

Aha, yeah, put it next to this.

Reply:
This still requires adding a global, and I don't feel `BattleTextParser` is the appropriate place for this because the logic there seems more related to battle protocol parsing, not generic crap like this, but OK.

Reply:
I mean, it's adding it to an existing global. Which I prefer to making a new global. Namespacing is useful!

And `@` in names is part of the battle protocol, because `|j|` etc are part of the battle protocol.